core-v1/src/main/scala/dsl/WrapImplicits.scala

```diff
@@ -0,0 +1,80 @@
+package com.rallyhealth.vapors.v1
+
+package dsl
+
+import cats.Functor
+import cats.implicits._
+import com.rallyhealth.vapors.v1.algebra.{SelectOutputType, WrapConst}
+
+trait WrapImplicits {
+  self: DslTypes with LowPriorityWrapImplicits =>
+
+  implicit def constFunctor[C[_] : Functor, O : OP](
+    implicit
+    aot: ConstOutputType[W, O],
+  ): ConstOutputType.Aux[W, C[O], C[aot.Out]]
+
+  implicit def selectFunctor[C[_] : Functor, I : OP, O : OP](
+    implicit
+    aot: SelectOutputType[W, I, O],
+  ): SelectOutputType.Aux[W, I, C[O], C[aot.Out]]
+
+}
+
+trait LowPriorityWrapImplicits {
+  self: DslTypes =>
+
+  implicit def constId[O : OP]: ConstOutputType.Aux[W, O, W[O]]
+
+  implicit def selectId[I : OP, O : OP]: SelectOutputType.Aux[W, I, O, W[O]]
+
+  /**
+    * Use this to implement all the DSL definitions
+    */
+  protected def defn: WrapDefinitions[W, OP]
```

Comment:
This comment is a little unclear... I should explain what I mean by "DSL definitions"